public/app/features/dashboard/components/TransformationsEditor/TransformationsEditor.tsx

```diff
@@ -125,6 +146,20 @@ class UnThemedTransformationsEditor extends React.PureComponent<TransformationsE
     }
   }
 
+  componentDidUpdate(prevProps: Readonly<TransformationsEditorProps>, prevState: Readonly<State>): void {
+    if (config.featureToggles.transformationsRedesign) {
+      const prevHasTransforms = prevState.transformations.length > 0;
+      const prevShowPicker = !prevHasTransforms || prevState.showPicker;
+
+      const currentHasTransforms = this.state.transformations.length > 0;
+      const currentShowPicker = !currentHasTransforms || this.state.showPicker;
+
+      if (prevShowPicker !== currentShowPicker) {
+        this.setState({ scrollTop: currentShowPicker ? Math.random() / 2 : 1000000 });
```

Comment:
Can we document and/or add variables for any numeric constants?

Reply:
done